### ISSUE TITLE
factory constructor of entity causing Query returns abnormal result

### ISSUE THREAD

**Issue by xiaozhizhong** [pinchbv/floor#724]
When a Entity has more than one constructors, and a factory constructor is defined ahead of other constructor in the Entity class.
For example:
```
@entity
class Task {
  @PrimaryKey(autoGenerate: true)
  final int? id;

  final String? message;

  final TaskType? type;
  
  /// This factory constructor cause problems!!!!
  factory Task.combine(Task other){
     return copyWith(message: "$message,${other.message}")
  }

  Task({this.id, this.isRead, this.message, this.timestamp, this.type});

  //other codes
}
```
And Query task like 
```
@Query('SELECT * FROM task WHERE type = :type')
  Future<Task?> findTaskByType(TaskType type);
```
The generated codes of this Query will be like:
```
return _queryAdapter.query(
        'SElECT * FROM Task WHERE type= ?',
        mapper: (Map<String, Object?> row) => Task(),  /// mapper creating Task without any params!!!
        arguments: [
        //other codes
```
Then run the app, `findTaskByType` will return a nonNull task but all of its properties are null.

If I swape the position of two construtors
```
@entity
class Task {
  @PrimaryKey(autoGenerate: true)
  final int? id;

  final String? message;

  final TaskType? type;
  
  Task({this.id, this.isRead, this.message, this.timestamp, this.type});

  /// This factory constructor is moved down
  factory Task.combine(Task other){
     return copyWith(message: "$message,${other.message}")
  }

  //other codes
}
```
Then the generated codes  is correct! And I can get the correct task when run the Query!

**Comment by dkaera** [comment]
@xiaozhizhong 
yup, Floor generator takes only the first constructor. I guess it was designed this way because Entity models are usually used only in Data layer.
We can improve it by looking for an appropriate constructor that fills up all the fields except those marked as ignored.

**Comment by dkaera** [comment]
@xiaozhizhong 
Issue solved in PR #734 
By the way dart lints highlights to [avoid placing named constructors before unnamed one](https://dart-lang.github.io/linter/lints/sort_unnamed_constructors_first.html)
